app/main.py

```diff
@@ -1 +1,31 @@
-# write your code here
+class Animal:
+    alive = []
+
+    def __init__(self, name: str, health: int = 100) -> None:
+        self.name = name
+        self.health = health
+        self.hidden = False
+        self.alive.append(self)
+
+    def die(self) -> None:
+        self.alive.remove(self)
+
+    def __repr__(self) -> str:
+        return (f"{{Name: {self.name}, "
+                f"Health: {self.health}, "
+                f"Hidden: {self.hidden}}}")
+
+
+class Herbivore(Animal):
+
+    def hide(self) -> None:
+        self.hidden = not self.hidden
+
+
+class Carnivore(Animal):
+
+    def bite(self, herbivore: object) -> None:
+        if isinstance(herbivore, Herbivore) and herbivore.hidden is False:
```

Comment:
There is no need to evaluate `hidden` attribute with is, rewrite as follows:
```suggestion
        if isinstance(herbivore, Herbivore) and not herbivore.hidden:
```